hot.h

```diff
@@ -0,0 +1,297 @@
+#ifndef INCLUDED_OPENTRACING_HOT_H
```

Comment:
Disclaimer: I did just some of `Span`, `Context` and `Tracer`. It's not a complete implementation.